asciidoc/volume1/use-cases/tf1-ch-c-use-case-stad.adoc

```diff
@@ -214,6 +225,163 @@ NOTE: This requirement supplements RR1162 in <<ref_ieee_11073_10700_2022>>: _The
+
+NOTE: two distinct epochs are created by a non-slewing time adjustment, each with a distinct time-reference frame. Both the rate of the passage of time and the determination time assigned to a single event may differ significantly between epochs (from the perspective of an external observer). 
+
+NOTE: non-slewing time adjustments may occur, for example, when a device rejoins a network, an absent <<acronym_ts_service>> returns to operation or be caused by hardware failure or operator error (e.g., making non-slewing adjustments to the <<acronym_ts_service>> time-reference frame while it is being used by one or more <<vol1_spec_sdpi_p_actor_somds_participant>>s). 
+
+NOTE: non-slewing time adjustments may result in a constant or variable offset between epochs. For constant offsets, the difference (to an unbiased observer) between any two timestamps obtained in different epochs is constant. For variable offsets, the difference (to an unbiased observer) between any two timestamps obtained in different epochs depends on when, within each epoch, the timestamp was obtained. 
+
+====== Safety, Effectiveness & Security Considerations and Requirements
+
+// This provides information for auditing. 
+.R1560
+[sdpi_requirement#r1560,sdpi_req_level=shall]
+****
+The <<vol1_spec_sdpi_p_actor_somds_participant>> shall include the determination time of the log entry in both the time-reference frame before, and after, each non-slewing clock adjustment. 
+
+.Notes
+[%collapsible]
+====
+
+NOTE: This requirement supplements TR1340 in <<ref_ieee_11073_10700_2022>>: _An SDC BASE PARTICIPANT SHOULD log each non-slewing adjustment of the local clock._
+
+====
+****
+
+// This is for providers to inform consumers of the non-slewing adjustment.
+// It is necessary to have a version here for providers that don't use NTP clock-discipline to smoothly adjust clocks and just set the clock (hopefully not going back in time).
+// Using `ClockState/@LastSet` like this avoids having to extend everything that needs a timestamp to support versioning (because any timestamp in the MDIB before the LastSet
+// is questionable following a transition to a new epoch). Epoch versioning is then an extension that lets the consumer determine how questionable a timestamp is. 
+// If we have a `Epochs/@Current` and update `ClockState/@LastSet` I don't think we need to also include a "Questionable" flag or change `ClockState/@ActiviationState` as proposed
+// during the workshop. Using `ClockState/@LastSet` seems better than just changing the @Activation state because the consumer could determine which timestamps are questionable.
+.R1522
+[sdpi_requirement#r1522,sdpi_req_level=shall]
+****
+When the <<vol1_spec_sdpi_p_actor_somds_provider>> detects a step adjustment of a system clock, used in making its System Function Contribution (<<acronym_sfc>>), the <<vol1_spec_sdpi_p_actor_somds_provider>> shall either:
+
+* initiate a new MDIB sequence by assigning a new MDIB sequence identifier, or
```

Comment:
This list does not cover the option of not setting a new sequence id, but just provding a failed clock.

Reply:
That's right. My thinking was setting `pm:ClockState/@LastSet` and updating the `sdpi:Epochs/@Version` was all that was necessary to communicate the problem with the sync methods allowed. However, this requires the extension. I've added it as #2:

1. initiate a new MDIB sequence by assigning a new MDIB sequence identifier, or
1. set `pm:ClockState/@ActivationState` to `StndBy` when any timestamp in a <<acronym_mdib>> version was not obtained from the time-reference frame of the active clock in the same version, or 
1. set `pm:ClockState/@LastSet` to the earliest time that is unambiguously in the current epoch and increment `sdpi:Epochs/@Version`.

I'm not sure option 2 is worthwhile though because it means leaving the `@ActivationState` as not "On" until any infrequently updated item (e.g., patient, location context etc) gets a new timestamp, which may be many  hours. Still, it could be a useful option for some  providers and I have no objection to it. 

I'm suggesting `StndBy` instead of `Fail` here so that participants could use `Fail` when their clock itself was faulty. 10207:§B.28 describes `StndBy` as "The component is ready to be operated but not currently operating", which seems true of the time-stamps the clock component has created so makes sense to me. `StndBy` could also be used for clocks waiting to synchronize, however they also have `ClockState/@RemoteSync` and `NotRdy` to indicate initialization.

Reply:
My line of thought was: effectively, the clock has failed working due to non-slewing adjustment such that timestamps in the MDIB are not coherent anymore. As long as we can argue "there is a fallback for consumers/providers supporting versioned epochs" or "A failed clock means: do not trust timestamps in the MDIB", we are just fine.

StandBy is also fine with me. We only have to make sure that this flag unambigiously tells consumers that timestamps are not trustworthy. Maybe this implication can be added as note to the requirement.

Reply:
Btw. this is not a logical OR over bullet points 1, 2 and 3, but a logical

`1 or 2 or (2 and 3)`

The latter case is for consumers who do not understand epoch versions.

Reply:
Yep. So this supports the case where consumers understand that `ClockState/@ActivationState=="StndBy"` but don't want to deal with `sdpi:Epochs/@Version`. With [changes above](https://github.com/IHE/DEV.SDPi/pull/354#discussion_r1893153083) R1522 captures `1 or 2 or (2 and 3)` by restating 2 in 3 for clarity.